### PR TITLE
presto_typeof - presto equivalent of pg_typeof function

### PR DESCRIPTION
**Pull request by fiedukow**
`presto_typeof` is meant to be used for implementing, manual testing and sql development.

The name of function uses `presto_` prefix to:
 1) Avoid possible clash of names in the future
 2) Follow `postgresql` standard for this kind of function

@dain - can you review?

### CODE REVIEW

**Comment by dain** [comment]
@martint can you review the signature?

**Comment by electrum** [comment]
I'd prefer a non-prefixed name like `typeof` or `typename`.

**Comment by fiedukow** [comment]
For me there is no difference I just thought the prefix is a better idea - but really not strong opinion there. If you'd like to change the name before merging just let me know - it's ~noop.

In a mean while, the acceptance/reject for rest of PR (including syntax itself) would be welcome.

**Comment by maciejgrzybek** [comment]
Any thoughts on this? It would be handy tool to test various stuff around type system while developing new features.

**Comment by martint** [comment]
I agree with @electrum on the naming. `typeof` seems reasonable to me.

I have some mild concerns about the semantics of this "function" (it's not really a function, since it operates on the type of expressions, not on values). In particular, the behavior is not well defined for things like:

```
typeof(x -> x + 1)                     -- when lambda expression get in
typeof(1 / 0)                          -- it would fail with division by 0, even though the type of the expression is INTEGER
typeof((VALUES (1, 'a'), (2, 'b')))    -- it should be something like multiset(row(integer, varchar(1)))
```

**Comment by maciejgrzybek** [comment]
@martint on the other hand if we want to tackle this in more generic way it may become much bigger and complex solution, I believe. But what's good about this patch is fact we can use it already and at some point tackle the broader problem (with solution to expressions you've given as an example).
That's because the set of problems solvable by this patch is a subset of the other, more complex solution. And there is no expression which would work with current patch which should not work with "the proper solution".

To sum up: we could make our lives a bit easier now with such a tiny change to code compared to not having such a feature at all. And we can have a GH issue to extend this feature further in future.

**Comment by fiedukow** [comment]
Seems to be integrated. Thanks! :-)
https://github.com/prestodb/presto/pull/5844/commits/27152953c0c3c72dbb39dee4fe0a9372d800ab07

Can we close that?

**Comment by maciejgrzybek** [comment]
You probably meant https://github.com/prestodb/presto/commit/3f489c824522bc0de863b0f8a06efdcc3b28044f, @fiedukow.
Closing to lower the noise in PRs. Reopen if need to continue discussion.
